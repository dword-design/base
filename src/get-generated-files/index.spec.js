import { endent, keys } from '@dword-design/functions'
import { outputFile } from 'fs-extra'

import { Base } from './..'

export default {
  'package.json script sort order': async () => {
    await outputFile(
      'package.json',
      JSON.stringify({
        scripts: {
          dev: 'base dev',
          test: 'base test',
        },
      })
    )
    expect(new Base().getGeneratedFiles()['package.json']).toEqual(endent`
      {
        "scripts": {
          "dev": "base dev",
          "test": "base test"
        }
      }

    `)
  },
  works() {
    expect(new Base().getGeneratedFiles() |> keys).toMatchSnapshot(this)
  },
}
