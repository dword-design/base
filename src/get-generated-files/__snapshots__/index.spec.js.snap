// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index no metadata sync 1`] = `
Array [
  "babel.config.json",
  ".commitlintrc.json",
  ".cz.json",
  ".editorconfig",
  ".eslintrc.json",
  ".gitattributes",
  ".github/DEPRECATED_DEPENDENCIES_ISSUE_TEMPLATE.md",
  ".github/labels.yml",
  ".github/workflows/build.yml",
  ".github/workflows/deprecated-dependencies.yml",
  ".github/workflows/sync-labels.yml",
  ".gitignore",
  ".gitpod.Dockerfile",
  ".gitpod.yml",
  ".huskyrc.json",
  ".releaserc.json",
  ".renovaterc.json",
  ".vscode/settings.json",
  ".yarnrc.yml",
  "LICENSE.md",
  "README.md",
  "package.json",
]
`;

exports[`index works 1`] = `
Array [
  ".commitlintrc.json",
  ".cz.json",
  ".devcontainer/devcontainer.json",
  ".editorconfig",
  ".eslintrc.json",
  ".gitattributes",
  ".github/DEPRECATED_DEPENDENCIES_ISSUE_TEMPLATE.md",
  ".github/FUNDING.yml",
  ".github/labels.yml",
  ".github/workflows/build.yml",
  ".github/workflows/deprecated-dependencies.yml",
  ".github/workflows/sync-labels.yml",
  ".github/workflows/sync-metadata.yml",
  ".gitignore",
  ".gitpod.Dockerfile",
  ".gitpod.yml",
  ".releaserc.json",
  ".renovaterc.json",
  ".vscode/settings.json",
  "LICENSE.md",
  "README.md",
  "babel.config.json",
  "package.json",
]
`;
